frontend/src/components/SignupForm/SignupForm.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignupForm.css'
+import { UserContext } from '../../UserContext.js';
+
+const SignupForm = () => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```

Comment:
Another potential advanced feature opportunity:
1. Improve password security by salt and hashing (if you haven't looked into it).